packages/engine-ui/src/components/PlanBookingDetails.tsx

```diff
@@ -0,0 +1,205 @@
+  const { dispatch } = React.useContext(UIStateContext)
+  const history = useHistory()
+  const { routeId, bookingId } = useParams<{
+    routeId: string
+    bookingId: string
+  }>()
+
+  React.useEffect(
+    () => () => {
+      dispatch({ type: 'highlightBooking', payload: undefined })
+    },
+    [dispatch]
+  )
+
+  const [booking] = bookings.filter((d: any) => d.id === bookingId)
+
+  const handleDeleteClick = (bookingId: string) => {
+    if (
+      window.confirm(
+        'Är du säker på att du vill ta bort bokningen från den föreslagna planen?'
+      )
+    ) {
+      //function for removing a booking from a plan goes here.
+      return history.push(`/plans/routes/${routeId}`)
+    }
+  }
+
+  if (!booking || !bookings) return <p>Laddar bokning...</p>
+
+  return (
+    <MainRouteLayout redirect={`/plans/routes/${routeId}`}>
+      <Elements.Layout.Container>
+        <Elements.Layout.FlexRowWrapper>
+          <h3>Bokning</h3>
+          <Elements.Typography.RoundedLabelDisplay margin="0 0.5rem">
+            {helpers.withoutLastFourChars(bookingId)}
+            <Elements.Typography.SpanBold>
+              {helpers.getLastFourChars(bookingId)}
+            </Elements.Typography.SpanBold>
+          </Elements.Typography.RoundedLabelDisplay>
+        </Elements.Layout.FlexRowWrapper>
+        <Elements.Layout.SectionWithMargin>
+          {booking.metadata.cargo && (
+            <CapitalizeParagraph>
+              <Elements.Typography.SpanBold>
+                Innehåll:{' '}
+              </Elements.Typography.SpanBold>
+              {booking.metadata.cargo}
+            </CapitalizeParagraph>
+          )}
+          <Paragraph>
+            <Elements.Typography.SpanBold>
+              Ömtåligt:{' '}
+            </Elements.Typography.SpanBold>
+            {booking.metadata.fragile ? 'Ja' : 'Nej'}
+          </Paragraph>
+          {booking.size.measurement && (
+            <Paragraph>
+              <Elements.Typography.SpanBold>
+                Mått:{' '}
+              </Elements.Typography.SpanBold>
+              {booking.size.measurement.map((item: number, index: number) =>
+                booking.size.measurement.length === index + 1
+                  ? `${item} cm `
+                  : `${item}x`
+              )}
+            </Paragraph>
+          )}
+          {booking.size.weight && (
+            <Paragraph>
+              <Elements.Typography.SpanBold>
+                Vikt:{' '}
+              </Elements.Typography.SpanBold>
+              {`${booking.size.weight} kg`}
+            </Paragraph>
+          )}
+        </Elements.Layout.SectionWithMargin>
+        <Elements.Layout.SectionWithMargin>
+          <Elements.Layout.MarginBottomContainer>
+            <Elements.Typography.StrongParagraph>
+              Upphämtning
+            </Elements.Typography.StrongParagraph>
+            <CapitalizeParagraph>{booking.pickup.street}</CapitalizeParagraph>
+            {booking.pickup.time_windows &&
```

Comment:
time_windows isn't optional, right?

Reply:
It is for a booking

Reply:
Ok, then that should be reflected in typings as well